src/main/java/graphql/validation/rules/DeferDirectiveOnValidOperation.java

```diff
@@ -0,0 +1,56 @@
+package graphql.validation.rules;
+
+import graphql.language.BooleanValue;
+import graphql.language.Directive;
+import graphql.language.Document;
+import graphql.language.Node;
+import graphql.language.OperationDefinition;
+import graphql.validation.AbstractRule;
+import graphql.validation.ValidationContext;
+import graphql.validation.ValidationErrorCollector;
+import java.util.List;
+import java.util.Optional;
+
+import static graphql.language.OperationDefinition.Operation.SUBSCRIPTION;
+import static graphql.validation.ValidationErrorType.MisplacedDirective;
+
+/**
+ * Defer Directive is Used On Valid Operations
+ *
+ * A GraphQL document is only valid if defer directives are not used on subscription types.
+ */
+public class DeferDirectiveOnValidOperation extends AbstractRule {
+
+    public DeferDirectiveOnValidOperation(ValidationContext validationContext, ValidationErrorCollector validationErrorCollector) {
+        super(validationContext, validationErrorCollector);
+        this.setVisitFragmentSpreads(true);
+    }
+
+    @Override
+    public void checkDirective(Directive directive, List<Node> ancestors) {
+        if (!directive.getName().equals("defer") ||
+                (directive.getArgumentsByName().get("if") != null && !((BooleanValue) directive.getArgumentsByName().get("if").getValue()).isValue() )) {
+            return;
+        }
```

Comment:
I am not sure we can use the `if` attr here - we dont know if its a variable value in validation and `$varName` is valid on the if.  This only handles AST booleans and this is not correct enough

Reply:
This needs to be fixed

Reply:
Added ifArgumentMightBeFalse